File: models/Commande.java
package models;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Commande {
    private int id;
    private int client;
    private List<Burger> burgers = new ArrayList<>();
    private List<Menu> menus = new ArrayList<>();
    private List<Complement> complements = new ArrayList<>();
    private EtatCommande etat = EtatCommande.EN_COURS;
    private TypeCommande type;
    private LocalDate date = LocalDate.now();
    private boolean payee = false;

    public Commande(int id, int client, TypeCommande type) {
        this.id = id;
        this.client = client;
        this.type = type;
    }

    public double calculerTotal() {
        double total = 0;
        for (Burger b : burgers) total += b.getPrix();
        for (Menu m : menus)total += m.getPrix();
        for (Complement c : complements) total += c.getPrix();
        return total;
    }

    public void payer() {
        if (!payee) {
            payee = true;
            etat = EtatCommande.VALIDE;
        }
    }

    public EtatCommande getEtat() { return etat; }
    public LocalDate getDate() { return date; }
}

File: Models/Commande.cs
using System.ComponentModel.DataAnnotations;

namespace BrasilBurger.Models
{
    public enum TypeCommande
    {
        SurPlace,
        Recuperer,
        Livrer
    }

    public enum EtatCommande
    {
        EnCours,
        Pret,
        Termine,
        Annule
    }

    public class Commande
    {
        public int Id { get; set; }

        [Required]
        public int ClientId { get; set; }

        public Client? Client { get; set; }

        public DateTime Date { get; set; } = DateTime.Now;

        [Required]
        public TypeCommande Type { get; set; }

        public EtatCommande Etat { get; set; } = EtatCommande.EnCours;

        public decimal Total { get; set; }

        public int? ZoneId { get; set; }

        public Zone? Zone { get; set; }

        public int? LivreurId { get; set; }

        public Livreur? Livreur { get; set; }

        // Navigation
        public ICollection<CommandeItem> Items { get; set; } = new List<CommandeItem>();

        public Paiement? Paiement { get; set; }
    }
}